src/errorMiddleware.js

```diff
@@ -0,0 +1,22 @@
+module.exports = (hooks = {}) => async (error, req, res, next) => {
+  if (hooks.beforeSend) {
+    try {
+      await hooks.beforeSend(req, res)
+    } catch (beforeSendError) {
+      return next(beforeSendError)
```

Comment:
@hharnisc makes sense to have a try/catch block for the hook. I have a question about it: if we call `next(beforeSendError)`, we wouldn't be fulfilling the `buffer-rpc` error contract with the client, would we? I am curious to know if this an intentional design choice.